api/builder.go

```diff
@@ -6,103 +6,51 @@ import (
-	Number           uint64         `json:"blockNumber"   gencodec:"required"`
-	GasLimit         uint64         `json:"gasLimit"      gencodec:"required"`
-	GasUsed          uint64         `json:"gasUsed"       gencodec:"required"`
-	Timestamp        uint64         `json:"timestamp"     gencodec:"required"`
-	ExtraData        []byte         `json:"extraData"     gencodec:"required"`
-	BaseFeePerGas    *big.Int       `json:"baseFeePerGas" gencodec:"required"`
-	BlockHash        common.Hash    `json:"blockHash"     gencodec:"required"`
-	TransactionsRoot common.Hash    `json:"transactionsRoot"  gencodec:"required"`
-}
-
-// JSON type overrides for executableData.
-type executionPayloadHeaderMarshalling struct {
-	Number        hexutil.Uint64
-	GasLimit      hexutil.Uint64
-	GasUsed       hexutil.Uint64
-	Timestamp     hexutil.Uint64
-	BaseFeePerGas *hexutil.Big
-	ExtraData     hexutil.Bytes
-	LogsBloom     hexutil.Bytes
-}
-
-// See https://github.com/flashbots/mev-boost#signedblindedbeaconblock
-type SignedBlindedBeaconBlock struct {
-	Message   *BlindedBeaconBlock `json:"message"`
-	Signature string              `json:"signature"`
-}
-
-// See https://github.com/flashbots/mev-boost#blindedbeaconblock
-type BlindedBeaconBlock struct {
-	Body BlindedBeaconBlockBody `json:"body"`
-}
-
-type BlindedBeaconBlockBody struct {
-	ExecutionPayload ExecutionPayloadHeaderV1 `json:"execution_payload_header"`
-}
-
-type SignedBuilderReceipt struct {
-	Message   *BuilderReceipt `json:"message"`
-	Signature string          `json:"signature"`
-}
-
-type BuilderReceipt struct {
-	PayloadHeader    ExecutionPayloadHeaderV1 `json:"execution_payload_header"`
-	FeeRecipientDiff uint256.Int              `json:"feeRecipientDiff"`
-}
-
-func GetPayloadHeader(ctx context.Context, cl *rpc.Client, log logrus.Ext1FieldLogger, payloadId PayloadID) (*ExecutionPayloadHeaderV1, error) {
-	e := log.WithField("payload_id", payloadId)
-	e.Debug("getting payload")
-	var result ExecutionPayloadHeaderV1
-	err := cl.CallContext(ctx, &result, "builder_getPayloadHeaderV1", payloadId)
+	pubkey := "0xf9716c94aab536227804e859d15207aa7eaaacd839f39dcbdb5adc942842a8d2fb730f9f49fc719fdb86f1873e0ed1c2"
+	err := cl.CallContext(ctx, &result, "builder_getHeaderV1", "0x1", pubkey, blockHash.Hex())
 	if err != nil {
 		e = e.WithError(err)
 		if rpcErr, ok := err.(gethRpc.Error); ok {
 			code := ErrorCode(rpcErr.ErrorCode())
 			if code != UnavailablePayload {
-				e.WithField("code", code).Warn("unexpected error code in get-payload response")
+				e.WithField("code", code).Warn("unexpected error code in get-payload header response")
 			} else {
-				e.Warn("unavailable payload in get-payload request")
+				e.Warn("unavailable payload in get-payload header request")
 			}
 		} else {
 			e.Error("failed to get payload header")
 		}
 		return nil, err
 	}
 	e.Debug("Received payload")
-	return &result, nil
+	return &result.Message.Header, nil
```

Comment:
Should return either `message` and `signature` or it should verify `message` against `signature` and return `message` if successful.